### PR TITLE
HZN-436, HZN-437, HZN-517: Refactor Syslogd

### PR DESCRIPTION
**Pull request by soleger**
This refactoring cleans up the Syslogd code significantly and provides some new experimental implementations of the syslog listener.

### CODE REVIEW

**Comment by soleger** [comment]
It also contains some general cleanup: removal of unused OutageService classes from the web UI, removal of unused model-importer.properties files, and normalization of PropertyPlaceholderConfigurer beans in Spring contexts.

**Comment by soleger** [comment]
Note: please leave the branch open after this pull request is complete because I will continue to work there.

File: opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml [hunk 1]
```diff
@@ -5,8 +5,20 @@
...
+    <bean class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
+        <property name="systemPropertiesModeName" value="SYSTEM_PROPERTIES_MODE_OVERRIDE" />
+        <!-- 
+          We need to ignore unresolvable placeholders since if multiple PropertyPlaceholderConfigurer
+          preprocessors are in use in a single context (such as inside unit tests), Spring will encounter
+          the issue documented here:
+
+          https://jira.springsource.org/browse/SPR-6428
+
+          TODO: Fix this behavior after upgrading to Spring 3.1.
```
Comment (j-white) [inline opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml]: We should probably update these comments with the new reasoning for not using '<context:property-placeholder />'.

File: features/events/syslog/pom.xml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    </dependency>
+    <dependency>
+        <groupId>io.dropwizard.metrics</groupId>
+        <artifactId>metrics-core</artifactId>
+        <version>3.1.2</version>
+    </dependency>
+    <dependency>
+      <groupId>com.lmax</groupId>
+      <artifactId>disruptor</artifactId>
+      <version>3.3.2</version>
```
Comment (j-white) [inline features/events/syslog/pom.xml]: We use the disruptor in features/newts/pom.xml as well, we should probably move the version for a property.

File: features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/Syslogd.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    /**
+     * <p>onInit</p>
+     */
+    @Override
+    protected void onInit() {
+
+        try {
+            // clear out the known nodes
+            SyslogdIPMgrJDBCImpl.getInstance().dataSourceSync();
```
Comment (j-white) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/Syslogd.java]: Any reason we're not using the SyslogdIPMgrDaoImpl instead?

File: features/events/syslog/src/main/resources/META-INF/opennms/applicationContext-syslogDaemon.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  <bean id="syslogReceiverJavaNet" class="org.opennms.netmgt.syslogd.SyslogReceiverJavaNetImpl">
+    <constructor-arg ref="syslogdConfigFactory"/>
+  </bean>
+
+  <bean id="syslogReceiverCamelNetty" class="org.opennms.netmgt.syslogd.SyslogReceiverCamelNettyImpl">
+    <constructor-arg ref="syslogdConfigFactory"/>
+  </bean>
+
+  <!-- Expose one particular implementation as the chosen implementation -->
```
Comment (j-white) [inline features/events/syslog/src/main/resources/META-INF/opennms/applicationContext-syslogDaemon.xml]: Shouldn't we be setting a default here?

**Comment by j-white** [comment]
Overall it looks good, but OpenNMS doesn't start if syslogd is enabled.

File: features/events/syslog/src/main/resources/META-INF/opennms/applicationContext-syslogDaemon.xml (hunk 4 above)
Comment (soleger) [inline features/events/syslog/src/main/resources/META-INF/opennms/applicationContext-syslogDaemon.xml]: This was just used for some testing so far. I'll be refactoring this as we modularize things further for Minion. Work in progress! :)

File: features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/Syslogd.java (hunk 3 above)
Comment (soleger) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/Syslogd.java]: I think the DAO version was written by an intern we had and has not been fully tested yet. In any case, this known IP address list needs to be merged with the implementation in Trapd and provided as an OSGi service from the DAO project (opennms-dao). That's more refactoring that will be future work in this branch.

File: opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml (hunk 1 above)
Comment (soleger) [inline opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml]: I'll change these comments to reflect this.

**Comment by soleger** [comment]
Well, that's a blocker! :) Can you open a bug with the details of the failure (stack trace, etc)? Thanks.

**Comment by soleger** [comment]
OK, the startup issue should be resolved by the latest commits. Please review and merge after bamboo finishes: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS298

File: opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml (hunk 1 above)
Comment (soleger) [inline opennms-correlation/drools-correlation-engine/src/main/resources/META-INF/opennms/correlation-engine.xml]: I opened http://issues.opennms.org/browse/NMS-8032 about this.

**Pull request merged**